dynamic_programming/fibonacci_bottom_up.cpp

```diff
@@ -1,7 +1,41 @@
-#include <iostream>
-using namespace std;
+/**
+ * @file
+ * @brief Implementation to find n th fibonacci_number
+ * (https://en.wikipedia.org/wiki/Fibonacci_number)
+ *
+ * @details
+ * The Fibonacci sequence is a series of numbers where each number is the
+ * sum of the two preceding ones, starting with 0 and 1.
+ *
+ * ### Algorithm
+ * The idea of the algorithm is to find the ith fibonacci number.
+ * DP state : F(i) = ith fibonacci number
+ * DP state transition :
+ *      F(i) = F(i-1) + F(i-2)
+ * DP base case :
+ *      F(0) = 0
+ *      F(1) = 1
+ * Space reduction :
+ *  * for i the fibonacci store only i-1 th and i-2 th state.
+ * @author [Santo Cariotti](https://github.com/dcariotti)
+ * @author [tapasrnk](https://github.com/tapasrnk)
+ */
+#include <cassert>   /// for assert
+#include <iostream>  /// for IO operations
+#include <vector>    /// for storing the values of nth fibonacci number
+/**
+ * @namespace dynamic_programming
+ * @brief Dynamic Programming algorithms
+ */
+namespace dynamic_programming {
 int fib(int n) {
```

Comment:
This function should be documented explaining the use of its parameters, a brief description, and what it returns.